src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -451,38 +417,88 @@ internal TimePoint(ReadOnlySpan<byte> rawData, int intPtrSize)
             // The length of the string is stored at byte* + sizeof(IntPtr).
             private static unsafe string GetStringFromPointer(ref ReadOnlySpan<byte> rawData, ref int index, int intPtrSize)
             {
-                ulong stringLength;
-                ReadOnlySpan<byte> buffer;
+                string result;
+
                 if (intPtrSize == 4)  // 32 bit machine
                 {
-                    stringLength = MemoryMarshal.Read<uint>(rawData.Slice(index + intPtrSize));
                     IntPtr stringData = new IntPtr(MemoryMarshal.Read<int>(rawData.Slice(index)));
-                    buffer = new ReadOnlySpan<byte>(stringData.ToPointer(), (int)stringLength);
+                    result = PointerToString(stringData);
                 }
                 else // 64 bit machine
                 {
-                    stringLength = MemoryMarshal.Read<ulong>(rawData.Slice(index + intPtrSize));
                     IntPtr stringData = new IntPtr(MemoryMarshal.Read<long>(rawData.Slice(index)));
-                    buffer = new ReadOnlySpan<byte>(stringData.ToPointer(), (int)stringLength);
+                    result = PointerToString(stringData);
                 }
 
-                if (stringLength == 0)
-                {
-                    index += intPtrSize * 2;
-                    return string.Empty;
-                }
+                index += intPtrSize;
 
-                index += intPtrSize * 2;
-#if NETSTANDARD2_0
-                return Encoding.UTF8.GetString(buffer.ToArray());
-#else
-                return Encoding.UTF8.GetString(buffer);
-#endif
+                return result;
             }
 
         };
 
-        #endregion
+        #region Union
+        // The folowing structs/enums are to enable us to simulate Unions on the Native side
+
```

Comment:
Why are these unions instead of being just separate data types like all other transformers?

Reply:
This change was done on the native side several months ago, so I am not 100% positive on all the reasons. One idea though was to reduce the amount of interop functions. This also only has 2 possible input types on the Native side. Though aren't very good reasons, but again it was a while ago and we can't remember the exact reasons we went that route. We may, at some point in the future, go to variant types in the Native side, and if we do than this how we would do it.